### PR TITLE
[Merged by Bors] - Added missing docs to bevy_internal and added warn on missing docs

### PR DESCRIPTION
**Pull request by Sheepyhead**
# Objective
This contributes documentation that should cover the entirety of bevy_internal as requested in #3492 

## Solution
warn(missing_docs) has been activated and documentation has been added to missing parts so that no warnings appear from this setting

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
Extremely complex code, impeccable documented. Thanks! @mockersf, feel free to merge this one ASAP.

**Comment by alice-i-cecile** [comment]
As discussed on Discord, we should have a module level doc comment that explains that this is seperate from the `bevy` crate to ensure that dynamic linking works correctly.

**Comment by Sheepyhead** [comment]
> As discussed on Discord, we should have a module level doc comment that explains that this is seperate from the `bevy` crate to ensure that dynamic linking works correctly.

I added a line mentioning this, if a more in depth explanation is necessary I need someone who actually understands why this separation is necessary to help me writing it

File: crates/bevy_internal/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,8 @@
+#![warn(missing_docs)]
+//! This module contains plugins for quickly setting up a Bevy app, as well as publicizing of the subcrates Bevy contains
+//!
+//! This module is separated into its own crate to enable simple dynamic linking for Bevy
```
Comment (bjorn3) [inline crates/bevy_internal/src/lib.rs:4]: This an internal crate that shouldn't directly be used. The `bevy` crate should instead be used.

Comment (Sheepyhead) [reply crates/bevy_internal/src/lib.rs]: Thanks for pointing that out, I updated the crate doc to reflect this

**Pull request closed without merging**